iroha/src/block.rs

```diff
@@ -163,6 +163,14 @@ impl SignedBlock {
     pub fn verified_signatures(&self) -> Vec<Signature> {
         self.signatures.verified(&self.hash())
     }
+
+    /// Returns `SignedBlock` with verified signatures.
+    pub fn verify_signatures(mut self) -> SignedBlock {
```

Comment:
Maybe add new Block type `VerifiedBlock`? O make it impossible to have unverified signatures in `SignedBlock`?

Reply:
Actually I missed the fact that it is not needed to verify signatures in the places that we discussed. They are verified later along with the check of Roles at commit stage. So I removed this method.

Reply:
In fact we have already discussed it in another PR :smile: